common_knowledge/GitBranching.md

```diff
@@ -0,0 +1,45 @@
+
+# Git Branching
+
+We use a model something close to [Trunk-based development](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) 
+where we try to merge code to master often and send in small pull requests.
+
+We use the standard/default merge strategy for Git which avoids helps avoid
+merge conflicts so you can merge master and smaller (possibly stacked) PRs into
+your development branch often.
+
+# Key Suggestions
+
+## Commit early, commit often
+
+Make sure to commit your code early and often. This allows for smaller and
+discrete integration tests but also reduces the chance that your code will cause
+a merge conflict later.
+
+## Conventional Commits
+
+Use [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) in your PR title.
+
+## Send Smaller PRs
+
+Generally a PR should do one thing only.  Try to break up your larger PRs so that
+it's easy to review and merge into master quickly.
+
+If other developers are sending in larger PRs please recommend that they break
+them into smaller PRs. 
+
+## Avoid Rebase.
+
+Don't rebase your branches as this will increase the risk of merge conflicts.
+
+There’s really no major advantage to rebase other than a cleaner git history.
+However, the downsides are higher risk of merge conflicts and conflicts when you
+share rebased commits
+
+## Write Unit Tests
```

Comment:
I feel like we should leave this test section out for now unless we want to truly expand on the unit test vs lifecycle test debate. This recommendation does not apply to all situations (i.e. command/query lifecycles). Also, I don't necessarily think a unit test is sufficient to say that a PR does not need to be checked out given that we have very little coverage i.e. your unit test might pass but its use when integrated with other components may fail.

Reply:
@timolegros I'm going to leave this one out for now. The point I was trying to make is that if code is 100% covered by tests then you don't need to test it again.  Like if you have a merge sort and that's 100% tested then there's no need to re-test it (assuming your tests are proper)